great_expectations/expectations/metrics/map_metric_provider/map_metric_provider.py

```diff
@@ -0,0 +1,686 @@
+        if metric_name.endswith(metric_suffix):
+            has_aggregate_fn: bool = False
+
+            if execution_engine is not None:
+                try:
+                    _ = get_metric_provider(
+                        f"{metric_name}.{MetricPartialFunctionTypes.AGGREGATE_FN.metric_suffix}",
+                        execution_engine,
+                    )
+                    has_aggregate_fn = True
+                except gx_exceptions.MetricProviderError:
+                    pass
+
+            if has_aggregate_fn:
+                dependencies["metric_partial_fn"] = MetricConfiguration(
+                    metric_name=f"{metric_name}.{MetricPartialFunctionTypes.AGGREGATE_FN.metric_suffix}",
+                    metric_domain_kwargs=metric.metric_domain_kwargs,
+                    metric_value_kwargs=base_metric_value_kwargs,
+                )
+            else:
+                dependencies["unexpected_condition"] = MetricConfiguration(
+                    metric_name=f"{metric_name[:-len(metric_suffix)]}.{MetricPartialFunctionTypeSuffixes.CONDITION.value}",
+                    metric_domain_kwargs=metric.metric_domain_kwargs,
+                    metric_value_kwargs=base_metric_value_kwargs,
+                )
+
+        # MapMetric uses "condition" metric to build "unexpected_count.aggregate_fn" and other listed metrics as well.
+        unexpected_condition_dependent_metric_name_suffixes: List[str] = list(
+            filter(
+                lambda element: metric_name.endswith(element),
+                [
+                    f".{SummarizationMetricNameSuffixes.UNEXPECTED_COUNT.value}.{MetricPartialFunctionTypes.AGGREGATE_FN.metric_suffix}",
+                    f".{SummarizationMetricNameSuffixes.UNEXPECTED_VALUE_COUNTS.value}",
+                    f".{SummarizationMetricNameSuffixes.UNEXPECTED_INDEX_QUERY.value}",
+                    f".{SummarizationMetricNameSuffixes.UNEXPECTED_INDEX_LIST.value}",
+                    f".{SummarizationMetricNameSuffixes.FILTERED_ROW_COUNT.value}",
+                    f".{SummarizationMetricNameSuffixes.UNEXPECTED_VALUES.value}",
+                    f".{SummarizationMetricNameSuffixes.UNEXPECTED_ROWS.value}",
+                ],
+            )
+        )
+        if len(unexpected_condition_dependent_metric_name_suffixes) == 1:
+            metric_suffix = unexpected_condition_dependent_metric_name_suffixes[0]
+            if metric_name.endswith(metric_suffix):
+                dependencies["unexpected_condition"] = MetricConfiguration(
+                    metric_name=f"{metric_name[:-len(metric_suffix)]}.{MetricPartialFunctionTypeSuffixes.CONDITION.value}",
+                    metric_domain_kwargs=metric.metric_domain_kwargs,
+                    metric_value_kwargs=base_metric_value_kwargs,
+                )
+
+        return dependencies
+
+    @staticmethod
+    def is_sqlalchemy_metric_selectable(
```

Comment:
@abegong @Shinnnyshinshin @anthonyburdi -- I wonder if our ability to deprecate the method `is_sqlalchemy_metric_selectable()` is thanks to the effort aimed at supporting `SQLAlchemy-2.0`; regardless, if we still need this method, it should be converted to be "private" (this is an entirely internal utility).  Please let me know (happy to discuss).  Thank you.